deepchem/data/data_loader.py

```diff
@@ -437,6 +439,69 @@ def _featurize_shard(self, shard):
     return (X, np.ones(len(X), dtype=bool))
 
 
+class JsonLoader(DataLoader):
+  """
+  Creates `Dataset` objects from input json files. 
+
+  This class provides conveniences to load data from json files.
+  It's possible to directly featurize data from json files using
+  pandas, but this class may prove useful if you're processing
+  large json files that you don't want to manipulate directly in
+  memory.
+  
+  """
+
+  def __init__(self,
+               tasks: List[str],
+               smiles_field: Optional[str] = None,
```

Comment:
Should we add options to load other fields from the JSON? We could perhaps have something like `json_fields` as a keyword argument that specifies a dictionary of JSON arguments and their types? I'm thinking something like:
```
json_files = {"smiles": str, "weight"; float, ...}
```
Then that way we don't need to have explicit fields like `smiles_field` or `id_field`

Reply:
Should we even be limiting it to files that contain molecules represented as SMILES strings?  Could we make it general enough that the features can be any kind of string or array-like object?  The Featurizer will determine how they get interpreted.

Reply:
+1 for a general implementation! JSON would likely be useful for many applications

Reply:
Sounds good! If JSON is only going to be used for materials, it will be easier but more restrictive to substitute in two `structure_field` and `composition_field` arguments. If we want the loaders to be totally general, I'll stick with `json_fields` and look for those JSON arguments specifically.

I'll add a `_featurize_material_df` method that takes those column labels and featurizes, similar to `_featurize_smiles_df`.